homeassistant/components/zwave_js/api.py

```diff
@@ -597,11 +765,183 @@ async def websocket_validate_dsk_and_enter_pin(
+        )
+        return
+    await client.driver.controller.async_provision_smart_start_node(provisioning_info)
+    connection.send_result(msg[ID])
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/unprovision_smart_start_node",
+        vol.Required(ENTRY_ID): str,
+        vol.Exclusive(DSK, "input"): str,
+        vol.Exclusive(NODE_ID, "input"): int,
+    }
+)
+@websocket_api.async_response
+@async_handle_failed_command
+@async_get_entry
+async def websocket_unprovision_smart_start_node(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict,
+    entry: ConfigEntry,
+    client: Client,
+) -> None:
+    """Unprovision a smart start node."""
+    try:
+        cv.has_at_least_one_key(DSK, NODE_ID)(msg)
+    except vol.Invalid as err:
+        connection.send_error(
+            msg[ID],
+            ERR_INVALID_FORMAT,
+            err.args[0],
+        )
+        return
+    dsk_or_node_id = msg.get(DSK) or msg[NODE_ID]
+    await client.driver.controller.async_unprovision_smart_start_node(dsk_or_node_id)
+    connection.send_result(msg[ID])
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/get_provisioning_entries",
+        vol.Required(ENTRY_ID): str,
+    }
+)
+@websocket_api.async_response
+@async_handle_failed_command
+@async_get_entry
+async def websocket_get_provisioning_entries(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict,
+    entry: ConfigEntry,
+    client: Client,
+) -> None:
+    """Get provisioning entries (entries that have been pre-provisioned)."""
+    provisioning_entries = (
+        await client.driver.controller.async_get_provisioning_entries()
+    )
+    connection.send_result(
+        msg[ID], [dataclasses.asdict(entry) for entry in provisioning_entries]
+    )
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/parse_qr_code_string",
+        vol.Required(ENTRY_ID): str,
+        vol.Required(QR_CODE_STRING): QR_CODE_STRING_SCHEMA,
+    }
+)
+@websocket_api.async_response
+@async_handle_failed_command
+@async_get_entry
+async def websocket_parse_qr_code_string(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict,
+    entry: ConfigEntry,
+    client: Client,
+) -> None:
+    """Parse a QR Code String and return QRProvisioningInformation dict."""
+    qr_provisioning_information = await async_parse_qr_code_string(
+        client, msg[QR_CODE_STRING]
+    )
+    connection.send_result(msg[ID], dataclasses.asdict(qr_provisioning_information))
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/supports_feature",
+        vol.Required(ENTRY_ID): str,
+        vol.Required(FEATURE): vol.All(
+            vol.Coerce(int), vol.In(val.value for val in ZwaveFeature)
```

Comment:
Generators can only be exhausted once. Make sure it's a list.
```suggestion
            vol.Coerce(int), vol.In([val.value for val in ZwaveFeature])
```

Reply:
Actually, you can just convert it right away.
```suggestion
            vol.Coerce(int), ZwaveFeature
```

Because it will raise `ValueError` if it's invalid.